parser/src/ast/variable.rs

```diff
@@ -28,16 +28,16 @@ pub enum VarKind {
 
 /// A variable reference, prefixed with `$`.
 #[derive(Debug, Clone, PartialEq)]
-pub struct VarReference<'a> {
+pub struct VarReference {
     /// The name of the variable.
-    pub name: &'a str,
+    pub name: String,
 }
 
 /// A variable assignation.
 #[derive(Debug, Clone, PartialEq)]
 pub struct Assign<'a> {
     /// The identifier of the variable.
-    pub name: &'a str,
+    pub name: String,
```

Comment:
Variable names do not need to be owned `String`s, especially since user-defined variables have more name restrictions than special variables like `$0` or `$!`.